egs/librispeech/ASR/lightweight_ctc/model.py

```diff
@@ -0,0 +1,183 @@
+          expansion_rate:
+            Expansion rate of the bottleneck layers.
+          skip_add:
+            Use skip connect in the bottleneck layers.
+          rnn_dim:
+            If positive, add a GRU layer with rnn_dim as the output dimension.
+        """
+        super().__init__()
+
+        assert subsampling_factor == 4, 'Only subsampling = 4 supported.'
+        self.num_features = num_features
+        self.skip_add = skip_add
+        self.rnn_dim = rnn_dim
+        c = first_out_channels
+        self.conv_subsample = nn.Sequential(
+            nn.Conv2d(
+                in_channels=1,
+                out_channels=c,
+                kernel_size=3,
+                padding=1,
+            ),
+            nn.BatchNorm2d(c),
+            nn.ReLU6(),
+            nn.Conv2d(
+                in_channels=c,
+                out_channels=c*2,
+                kernel_size=3,
+                stride=2,
+                padding=1,
+            ),
+            nn.BatchNorm2d(c*2),
+            nn.ReLU6(),
+            nn.Conv2d(
+                in_channels=c*2,
+                out_channels=c*4,
+                kernel_size=3,
+                stride=2,
+                padding=1,
+            ),
+            nn.BatchNorm2d(c*4),
+            nn.ReLU6(),
+        )
+
+        w = ((num_features+1)//2 + 1) // 2
+        c *= 4
+        self.bottleneck_layers = nn.Sequential()
+        for _ in range(blocks):
+            self.bottleneck_layers.append(
+                Bottleneck(
+                    in_channels=c,
+                    out_channels=c*2,
+                    expansion_rate=expansion_rate,
+                    w_stride=2,
+                    skip_add=False,
+                )),
+            self.bottleneck_layers.append(
+                Bottleneck(
+                    in_channels=c*2,
+                    out_channels=c*2,
+                    expansion_rate=expansion_rate,
+                    skip_add=skip_add,
+                )),
+            self.bottleneck_layers.append(
+                Bottleneck(
+                    in_channels=c*2,
+                    out_channels=c*2,
+                    expansion_rate=expansion_rate,
+                    skip_add=skip_add,
+                )),
+            c *= 2
+            w = (w+1) // 2
+        if rnn_dim > 0:
+            self.rnn = nn.GRU(c*w, rnn_dim, batch_first=True)
+            self.linear = nn.Linear(rnn_dim, num_classes)
+        else:
+            self.linear = nn.Linear(c*w, num_classes)
+
+    def forward(self, x: Tensor) -> Tensor:
+        """
+        Args:
+          x: input feature, with shape (N, 1, T, F),
+             where F is the feature dimension.
+
+        Returns:
+          A tensor with shape (N, round(T / 4), num_classes)
+        """
+        assert x.shape[-1] == self.num_features, \
+            f"Number of features should be {self.num_features} \
+            instead of {x.shape[3]}"
```

Comment:
```suggestion
            instead of {x.shape[-1]}"
```

Please be consistent.